platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -161,24 +167,28 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
 
         then:
         verifyAll(receivedProblem(0)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
+            severity == Severity.WARNING
+            fqid == 'compilation:java:compiler-warn-redundant-cast'
+            contextualLabel == 'redundant cast to java.lang.String'
         }
         verifyAll(receivedProblem(1)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
+            severity == Severity.WARNING
+            fqid == 'compilation:java:compiler-warn-redundant-cast'
+            contextualLabel == 'redundant cast to java.lang.String'
         }
         verifyAll(receivedProblem(2)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
+            severity == Severity.WARNING
+            fqid == 'compilation:java:compiler-warn-redundant-cast'
+            contextualLabel == 'redundant cast to java.lang.String'
         }
         verifyAll(receivedProblem(3)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
+            severity == Severity.WARNING
+            fqid == 'compilation:java:compiler-warn-redundant-cast'
```

Comment:
❓ Does it make sense to have the severity in the id? And would it make sense to kick out the `compiler` prefix? So we'd end up with something like `compilation:java:redundant-cast`.

Reply:
IMO the ID is just an ID, and it's not an issue if it contains redundant information.
Even if I'm pretty certain that the prefix here is stable (e.g. JDK8's list of [compiler codes](https://github.com/openjdk/jdk/blob/fcbd7a77c40eec49b469a9a7d140678d5ed3076a/langtools/src/share/classes/com/sun/tools/javac/resources/compiler.properties)), I would still avoid making an assumption of the code format (apart from the dot separation).

You have something in mind, where the shorter ID might be beneficial?

Reply:
Fine with me. Will some humans end up reading the ID at some point?

Reply:
In Buildship we will soon have a UI for problems. Here's a [screenshot](https://private-user-images.githubusercontent.com/419883/338255346-5597a534-9dc8-4122-92c7-a25af03b275c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDA1NDcsIm5iZiI6MTcxODY0MDI0NywicGF0aCI6Ii80MTk4ODMvMzM4MjU1MzQ2LTU1OTdhNTM0LTlkYzgtNDEyMi05MmM3LWEyNWFmMDNiMjc1Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxN1QxNjA0MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNTk2Y2VhNGE1ZDMwMGRkMTJmOTY1YjM5NWJlY2FjMGU2MmVkZjA0M2FhOGNlYWRkZDk4ZDRkYzliZGQzNWY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.d9Kd9T1Q9pvAjAVMHohb5k4QBTUylE_vZ7x0zygifIM) from my [wip pr](https://github.com/eclipse/buildship/pull/1306).